### ISSUE TITLE
Lua Version of JWLuaMenu plugin

### ISSUE THREAD

**Issue by rpatters1** [finale-lua/lua-scripts#449]
**Background**

RGP Lua 0.66 will include some features that make it possible to rewrite the JWLuaMenu plugin as a script. I would like to do this in the interests of making it possible for others to contribute to it or modify it to suit their needs. The new features are:

- An option that allows a script to run when Finale starts up (and only then).
- A new `finenv.GetFinaleMainWindow()` function that is required on Windows for manipulating menus.

The script will also depend on a new version of [luaosutils](https://github.com/finale-lua/luaosutils) that contains apis for interrogating and rearranging Finale's menus (actually, the menus of whatever process Lua is running in).

**The crux of the issue**

My question for this group is where to put the script on GitHub. The options would seem to be:

1. The src directory of this repo (or a subdirectory of it).
2. A separate repo in the finale-lua organization.
3. An entirely separate repo in my personal account.

The main challenge is the dependency on luaosutils. To add it to the src directory of this repo would mean accommodating luaosutils in the deployment process. I'm not sure how to do that. It is further complicated on macOS by the possibility that the luaosutils library has to be signed and notarized before a user could download and use it without an ugly process of blessing the library. I still am not sure of the answer to this question.

Like JWLuaMenu, the script will need a menu layout file. I am intending to use the same format for the script as the plugin.

**Comment by ThistleSifter** [comment]
My current thought is that in this repo would be good if possible, but I had a couple of quick questions.

Just thinking generally, are any other scripts (current or future) in this repo likely to utilize `luaosutils`? I guess the implied question I'm asking is: should we be looking at having the standard deployment process accommodate `luaosutils`?

And from there...:
- How should it be offered for download? Zip file, or a separate link for `luaosutils` (are there likely to be more libraries in this form used by other scripts in the future?) If a zip file, do instructions need to be included in the zip file or just on the download page?
- Any version management of `luaosutils`? If it is going to be potentially used by other scripts (either from here or somewhere else) what happens if a new version is overwritten with an old one? Current scripts avoid this sharing issue by being self contained in one file.

I'm not familiar with the notarizing process. How would that affect the way it is offered for download or where it is hosted?

**Comment by rpatters1** [comment]
`luaosutils` started out as a file downloading utility. The idea was that it could be used to check versions.

For the latest iteration, I have expanded it to include menu utilities and a process launcher that is silent. (`os.execute` is not silent on Windows: it opens a console window. While a macOS version is not strictly necessary, I added one for completeness. Both versions were generated for me in seconds by ChatGPT, which I am gradually finding to be indispensable.)

It occurred to me that an easy way to deploy it would be to pre-load it in RGP Lua, like `luasocket`. But that would defeat the goal of a fully open-source library that is not dependent on the RGP Lua release schedule.

The signing and notarizing is the biggest potential issue. Jan Angermüller has been using it in Perfect Layout. He may have a better handle on whether it is actually an issue.
